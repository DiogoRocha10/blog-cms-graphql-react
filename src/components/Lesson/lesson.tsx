import { CheckCircle, Lock } from "phosphor-react";
import { isPast, format } from "date-fns";
import ptBR from "date-fns/locale/pt-BR";
import styles from "./lesson.module.css";

interface LessonProps {
  title: string;
  slug: string;
  availableAt: Date;
  type: string;
}

export function Lesson(props: LessonProps) {
  const isLessonAvailable = isPast(props.availableAt);
  const availableAtFormatted = format(
    props.availableAt,
    "EEEE' - 'd' de 'MMMM' - 'k'h'mm",
    { locale: ptBR }
  );

  return (
    <a href={props.slug}>
      <span className={styles["date"]}>{availableAtFormatted}</span>
      <div className={styles["container"]}>
        <header className={styles["header"]}>
          {isLessonAvailable ? (
            <span className={styles["icon-check"]}>
              <CheckCircle size={20} />
              Conteúdo liberado
            </span>
          ) : (
            <span className={styles["icon-lock"]}>
              <Lock size={20} />
              Em breve
            </span>
          )}
          <span className={styles["type"]}>
            {props.type === "live " ? "AO VIVO" : "AULA PRÁTICA"}
          </span>
        </header>
        <strong className={styles["title"]}>{props.title}</strong>
      </div>
    </a>
  );
}
