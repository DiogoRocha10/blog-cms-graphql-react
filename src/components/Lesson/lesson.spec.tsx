import { Lesson } from "./lesson";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Lesson
        title="Aula 01"
        slug="aula-01"
        availableAt={new Date("2020-01-01")}
        type="live"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
