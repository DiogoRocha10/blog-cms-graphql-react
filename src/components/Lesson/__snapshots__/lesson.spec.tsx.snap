// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="aula-01"
>
  <span
    className="date"
  >
    terça-feira - 31 de dezembro - 21h00
  </span>
  <div
    className="container"
  >
    <header
      className="header"
    >
      <span
        className="icon-check"
      >
        <svg
          fill="currentColor"
          height={20}
          viewBox="0 0 256 256"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <polyline
            fill="none"
            points="172 104 113.3 160 84 132"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
          />
          <circle
            cx="128"
            cy="128"
            fill="none"
            r="96"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
          />
        </svg>
        Conteúdo liberado
      </span>
      <span
        className="type"
      >
        AULA PRÁTICA
      </span>
    </header>
    <strong
      className="title"
    >
      Aula 01
    </strong>
  </div>
</a>
`;
